extensions/read-marker.md

```diff
@@ -0,0 +1,145 @@
+
+These read markers mean that the actual user has read the message. This specification is *not* about message delivery at the client socket level.
+
+The read markers sent by a user MUST NOT be disclosed to other users by the server without an explicit opt-in.
+
+## Implementation
+
+The `read-marker` extension uses the `draft/read-marker` capability and introduces a new command, `MARKREAD`.
+
+The `draft/read-marker` capability MAY be negotiated, and affects which messages are sent by the server as specified below.
+
+Clients can know whether a user has already read newly received messages. For clients that display notifications about new messages or highlights, knowing when messages have been read can enable them to clear notifications for messages that were already read on another device.
+
+Clients never have to actively get the read timestamp because it is provided to them on join and as updated by the server, except for user targets where they have to request the initial read timestamp by sending a `MARKREAD` client get command.
+
+### `MARKREAD` Command
+
+The `MARKREAD` command can be sent by both clients and servers.
+
+This command has the following general syntax:
+
+    MARKREAD <target> [<timestamp>]
+
+The `target` parameter specifies a single buffer (channel or nickname).
+
+The `timestamp` parameter, if specified, MUST be a literal `*`, or have the format `timestamp=YYYY-MM-DDThh:mm:ss.sssZ`, as in the [server-time](https://ircv3.net/specs/extensions/server-time) extension.
+
+#### `MARKREAD` client set command
+
+    MARKREAD <target> <timestamp>
+
+When sent from a client, this command signals to the server that the last message read by the user, to the best knowledge of the client, has the specified timestamp. The timestamp MUST correspond to a previous message `time` tag. The timestamp MUST NOT be a literal `*`.
+
+The server MUST reply to a successful `MARKREAD` client set command using a `MARKREAD` server command, or using an error message.
+
+#### `MARKREAD` client get command
+
+    MARKREAD <target>
+
+When sent from a client, this `MARKREAD` command requests the server about the timestamp of the last message read by the user.
+
+The server MUST reply to a successful `MARKREAD` get command using a `MARKREAD` server command, or using an error message.
+
+#### `MARKREAD` server command
+
+When sent from a server, the `MARKREAD` command signals to the client that the last message read by the user, to the best knowledge of the server, has the specified timestamp. In that case, the command has the following syntax:
+
+    <prefix> MARKREAD <target> <timestamp>
+
+The `prefix` is the prefix of the client the message is sent to.
```

Comment:
Just noticed this:

1. `<prefix>` is syntactically the source/prefix of the IRC message, correct? e.g. `:n!u@h MARKREAD #test 2022-03-31T01:32:21.380Z`
2. I don't understand the purpose of specifying the prefix here. The prefix typically indicates the origin of the message, not the recipient, right? My sense is that this should not be specified, and server implementations should probably use the server name as the source/prefix of these messages.

Reply:
Originally this spec was introducing a `READ` command, and we figured that the same command might get re-used by a future spec to send read receipts between users. In that case it would've made sense in the new spec to allow the source/prefix to be different from the current user's. Note, other commands like `JOIN` indicate success by having the server reply with a `JOIN` message with the current user's source/prefix.

Although I agree it's a bit premature at this point, we're not even sure it's a good idea to re-use the same message for read receipts. I'll drop this requirement.

Reply:
That makes sense, thanks. I support dropping the requirement for now.

My sense is that read receipts would be better implemented as TAGMSG.